From: Jordan Petridis <jpetridis@gnome.org>
Date: Wed, 26 Nov 2025 16:07:57 +0200
Subject: parental-controls: Adapt for malcontent 0.14

libmalcontnet now has its own abstraction over users
and thus we need to pass the correct type, instead
of using ACT_ ones.

See [1] and [2]

[1] https://gitlab.freedesktop.org/pwithnall/malcontent/-/merge_requests/212
[2] https://gitlab.freedesktop.org/pwithnall/malcontent/-/commit/385af6bf636141007e8cd9da6066aa98238156c2

Part-of: <https://gitlab.gnome.org/GNOME/gnome-initial-setup/-/merge_requests/304>

Origin: https://gitlab.gnome.org/GNOME/gnome-initial-setup/-/commit/e3e8a12454
---
 .../pages/parental-controls/gis-parental-controls-page.c            | 6 ++++++
 meson.build                                                         | 5 +++++
 2 files changed, 11 insertions(+)

diff --git a/gnome-initial-setup/pages/parental-controls/gis-parental-controls-page.c b/gnome-initial-setup/pages/parental-controls/gis-parental-controls-page.c
index 07f44d1..239322e 100644
--- a/gnome-initial-setup/pages/parental-controls/gis-parental-controls-page.c
+++ b/gnome-initial-setup/pages/parental-controls/gis-parental-controls-page.c
@@ -156,8 +156,14 @@ gis_parental_controls_page_constructed (GObject *object)
 
   /* Set up the user controls. We can’t set #MctUserControls:user because
    * there’s no way to represent a not-yet-created user using an #ActUser. */
+#ifdef HAVE_MALCONTENT_0_14
+  mct_user_controls_set_user_account_type (MCT_USER_CONTROLS (page->user_controls),
+                                           MCT_USER_TYPE_CHILD);
+#else
   mct_user_controls_set_user_account_type (MCT_USER_CONTROLS (page->user_controls),
                                            ACT_USER_ACCOUNT_TYPE_STANDARD);
+#endif
+
   update_user_controls (page);
 
   app_filter = mct_app_filter_builder_end (&builder);
diff --git a/meson.build b/meson.build
index 55f544a..016ece1 100644
--- a/meson.build
+++ b/meson.build
@@ -75,9 +75,14 @@ libmalcontent_dep = dependency ('malcontent-0',
 libmalcontent_ui_dep = dependency ('malcontent-ui-1',
                                    version: '>= 0.11.0',
                                    required: get_option('parental_controls'))
+
 have_parental_controls = libmalcontent_dep.found() and libmalcontent_ui_dep.found()
 conf.set('HAVE_PARENTAL_CONTROLS', have_parental_controls)
 
+have_malcontent_014 = dependency ('malcontent-0', version: '>= 0.14.alpha', required: false).found()
+conf.set('HAVE_MALCONTENT_0_14', have_malcontent_014,
+         description: 'Defined to 1 if malcontent ≥ 0.14 is available')
+
 webkitgtk_dep = dependency('webkitgtk-6.0', required: get_option('webkitgtk'))
 have_webkitgtk = webkitgtk_dep.found()
 conf.set('HAVE_WEBKITGTK', have_webkitgtk)
